packages/core/src/middleware/koa-log.ts

```diff
@@ -41,14 +32,30 @@ export default function koaLog<StateT, ContextT, ResponseBodyT>(): MiddlewareTyp
   ResponseBodyT
 > {
   return async (ctx, next) => {
-    ctx.log = {};
+    // eslint-disable-next-line @silverhand/fp/no-let
+    let logType: Optional<LogType>;
+    // eslint-disable-next-line @silverhand/fp/no-let
+    let logPayload: LogPayload = {};
+
+    ctx.log = <T extends LogType, P extends LogPayloads[T]>(type: T, payload: P) => {
+      if (!logType) {
+        // eslint-disable-next-line @silverhand/fp/no-mutation
+        logType = type; // Use first initialized log type
+      }
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      logPayload = deepmerge(logPayload, payload);
```

Comment:
actually, do we need deep merge in our case? Shallow should be good enough?

Reply:
Usage: e.g.

```ts
ctx.log(type, { arr: [1]})
ctx.log(type, { arr: [2]})
// logPayload = { arr: [1, 2]}
```

Conclusion: keep it for now.

See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1650861621977389)